### ISSUE TITLE
TRRT error converter does no conversion (but it does execute on an error)

### ISSUE THREAD

**Issue by RieksJ** [tno-terminology-design/tev2-tools#43]
I have specified the following TRRT configuration: 

``` yaml
## Term Reference Resolution Tool
trrt:
  interpreter: default 
  converter: html-hovertext-link
  converter[error]: "{{showtext}}{{log 'TRRT error converter:' err.dir '/' err.file '@' err.line ':' err.pos '[' ref.showtext ']' level='warn'}}"
```

I would expect that this configuration would have the following effect for any TermRef for which it could not find a corresponding MRG-entry:

- [x] it would write a log-text, stating the directory/file + line:pos + the showtext of the TermRef (which it does)
- [x] it would replace the Termref with its showtext (which it doesn't). Instead, it leaves the TermRef intact.

The idea of the `converter[error]` is that the TermRef will actually be replaced by what it says.

**Comment by Ca5e** [comment]
Interesting, I didn't know `log` was an option within Handlebars. The TRRT currently doesn't perform a replacement step if the replacement string is empty, in the case of the current error converter it is empty because the use of `{{showtext}}` instead of `{{ref.showtext}}`.

**Comment by RieksJ** [comment]
Yes, it is. In the (new) TEv2-page on [handlebar helper functions](https://tno-terminology-design.github.io/tev2-specifications/docs/specs/syntax/handlebars-helper-functions), I have included the list of helpers that are available by default (which refer to their documentation). Knowing about them - particularly `#each` - was instrumental for imlementing the feature for having definitions be associated with notes, which many glossaries have (and which TEv2 now also does).

Ok for your explanation. Can you make the TRRT (and HRGT) replace the TermRef/MRGRef (in case of an error) with the output of the `err.converter` if that is not the empty string?

**Comment by RieksJ** [comment]
is this issue related to #45 ?

**Comment by Ca5e** [comment]
> is this issue related to #45 ?

They are related in the sense that `{{showtext}}` was still being used instead of `{{ref.showtext}}`. I've changed that in [this](https://github.com/tno-terminology-design/tev2-specifications/commit/706de663e5e07a97b9cd8638c256b1a63af475db) commit. I believe this issue is now resolved with the publication of verson 1.1.2 of the tools.
